testtracer_test.go

```diff
@@ -39,10 +46,39 @@ func (n testTracer) StartSpanWithOptions(opts StartSpanOptions) Span {
 
 // Inject belongs to the Tracer interface.
 func (n testTracer) Inject(sp Span, format interface{}, carrier interface{}) error {
-	return nil
+	span := sp.(testSpan)
+	switch format {
+	case TextMap:
+		// Just for testing purposes... generally not a worthwhile thing to
+		// propagate.
+		carrier.(TextMapWriter).Add(testHTTPHeaderPrefix+"opname", span.OperationName)
+		carrier.(TextMapWriter).Add(testHTTPHeaderPrefix+"hasparent", strconv.FormatBool(span.HasParent))
+		return nil
+	}
+	return ErrUnsupportedFormat
 }
 
 // Join belongs to the Tracer interface.
 func (n testTracer) Join(operationName string, format interface{}, carrier interface{}) (Span, error) {
+	switch format {
+	case TextMap:
+		// Just for testing purposes... generally not a worthwhile thing to
+		// propagate.
+		rval := testSpan{}
+		err := carrier.(TextMapReader).ReadAll(func(key, val string) error {
+			switch strings.ToLower(key) {
```

Comment:
This is the sort of thing that's going to be a gotcha for anyone who naively uses `TextMapCarrier` with an http header implementation... Oh well.